### ISSUE TITLE
ECWolf crashes; bug reporter hangs or does not show up at all

### ISSUE THREAD

**Issue by Blzut3** [ECWolfEngine/ECWolf#186]
**[Original report](https://bitbucket.org/ecwolf/ecwolf/issues/186) by Belmondo (Bitbucket: [DenisBelmondo](https://bitbucket.org/DenisBelmondo), GitHub: [DenisBelmondo](https://github.com/DenisBelmondo)).**

----------------------------------------

I am not sure what's causing this issue at all. When the bug reporter window shows up, I can't click on the "details" tab because the window is frozen and I get the stock windows error message saying "ECWolf is not responding." The only bit of information I could gather was from the "summary" section of the window (pictured below)
![veryfatal.png](https://bitbucket.org/repo/Xg5g6/images/1139681760-veryfatal.png)

This only seems to happen with my custom-built mod which [I'll link here so you can take a peek at it just in case I've done something silly.](https://drive.google.com/file/d/0B3FoZTPDQFgKVDlxaEdiaVVHbkU/view?usp=sharing)

**Comment by Blzut3** [comment]
**Original comment by Belmondo (Bitbucket: [DenisBelmondo](https://bitbucket.org/DenisBelmondo), GitHub: [DenisBelmondo](https://github.com/DenisBelmondo)).**

----------------------------------------

 I forgot to note, this is the February 13th x64 build I snagged on the DRDTeam website.

EDIT: Picking up either type of ammo yields this error seemingly at random. The address ECWolf fails to read is always 0000000000000008, incidentally.

**Comment by Blzut3** [comment]
Yeah, essentially that means a NULL pointer read I'm thinking.  A little surprised it's not showing up on Linux.  Will have to try the Windows build later.

**Comment by Blzut3** [comment]
Seems to work fine on my Windows machine.  Are you able to give more precise reproduction instructions?

**Comment by Blzut3** [comment]
**Original comment by Belmondo (Bitbucket: [DenisBelmondo](https://bitbucket.org/DenisBelmondo), GitHub: [DenisBelmondo](https://github.com/DenisBelmondo)).**

----------------------------------------

Sure! I'm doubtful that anything during the mod's production process would have to do with these errors but I'll give them to you in explicit detail before getting to the actual error itself in the second paragraph just in case.

I don't think I've had this happen to me before, this is probably the furthest I've gotten with making an ECWolf mod and this time around I changed the development process altogether both to remedy the "SLADE 3 doesn't alphabetically sort directories" problem and just for a little extra layer of fun. ANYWAY, the entire mod is in a directory, which is then zipped with a .pk3 extension using the "zip" tool supplied with Cygwin (3.0). Then, I open a command prompt (Powershell mostly), and point to the directory the mod is located as a parameter thus letting ECWolf load it (as a lot of people do I'm sure)

This particular error occurs anywhere between 2 minutes within the game, or upstream of 10. I've never had it happen when I pick up a custom weapon to my knowledge, only custom ammo types (which are "Clip_" and "_Clip2") that are **dropped by enemies** (the underscores were intentional). As soon as BJ collides with one of these ammo items (which happens **seemingly exclusively when they are dropped by any enemy**, I don't remember it crashing when I pick a fresh one off the ground), the game crashes and most of the time it gives me the error message pictured above but in rare circumstances it closes without warning.

**Comment by Blzut3** [comment]
**Original comment by Belmondo (Bitbucket: [DenisBelmondo](https://bitbucket.org/DenisBelmondo), GitHub: [DenisBelmondo](https://github.com/DenisBelmondo)).**

----------------------------------------

Sorry, I should also reiterate that it doesn't happen when I pick the ammotypes up the first time I pick them up. It could be the 27th or the 94th (or whatever arbitrary number) magazine I've collected. So in other words, It happens seemingly at random.

**Comment by Blzut3** [comment]
**Original comment by Belmondo (Bitbucket: [DenisBelmondo](https://bitbucket.org/DenisBelmondo), GitHub: [DenisBelmondo](https://github.com/DenisBelmondo)).**

----------------------------------------

I finally reproduced it! It happens when I run out of ammo of either ammo type and pick up a magazine of the corresponding ammo type. Say if I ran out of 7.92 rounds after emptying the entire repository and I pick up a 7.92 magazine off the ground, that's when the game would crash. It only seems to happen under these circumstances and never happens if you don't empty your entire ammo pool beforehand. I tested this with the latest DRDTeam build (both x64 and x86) and the stable release and they both exhibit this behavior.
